drivers/adc/adc_rpi_pico.c

```diff
@@ -0,0 +1,352 @@
+			      const struct adc_sequence *sequence,
+			      struct k_poll_signal *async)
+{
+	struct adc_rpi_data *data = dev->data;
+	int err;
+
+	adc_context_lock(&data->ctx, async ? true : false, async);
+	err = adc_rpi_start_read(dev, sequence);
+	adc_context_release(&data->ctx, err);
+
+	return err;
+}
+
+static int adc_rpi_read(const struct device *dev,
+			const struct adc_sequence *sequence)
+{
+	return adc_rpi_read_async(dev, sequence, NULL);
+}
+
+static void adc_context_start_sampling(struct adc_context *ctx)
+{
+	struct adc_rpi_data *data = CONTAINER_OF(ctx, struct adc_rpi_data,
+						 ctx);
+	const struct adc_rpi_config *config = data->dev->config;
+
+	data->channels = ctx->sequence.channels;
+	data->repeat_buf = data->buf;
+
+	hw_clear_bits(&ADC_RPI_HW->fcs, ADC_FCS_OVER_BITS);
+	hw_clear_bits(&ADC_RPI_HW->fcs, ADC_FCS_UNDER_BITS);
+	hw_clear_bits(&ADC_RPI_HW->cs, ADC_CS_ERR_STICKY_BITS);
+
+	/* Find next channel and start conversion */
+	hw_write_masked(&ADC_RPI_HW->cs, (find_lsb_set(data->channels) - 1) << ADC_CS_AINSEL_LSB,
+			ADC_CS_AINSEL_BITS);
+	hw_set_bits(&ADC_RPI_HW->cs, ADC_CS_START_ONCE_BITS);
+}
+
+static void adc_context_update_buffer_pointer(struct adc_context *ctx,
+					      bool repeat_sampling)
+{
+	struct adc_rpi_data *data = CONTAINER_OF(ctx, struct adc_rpi_data,
+						 ctx);
+
+	if (repeat_sampling) {
+		data->buf = data->repeat_buf;
+	}
+}
+
+/**
+ * @brief Function called on init for each RaspberryPi Pico ADC device. It setups all
+ *        channels to return constant 0 mV and create acquisition thread.
+ *
+ * @param dev RaspberryPi Pico ADC device
+ *
+ * @return 0 on success
+ */
+static int adc_rpi_init(const struct device *dev)
+{
+	const struct adc_rpi_config *config = dev->config;
+	struct adc_rpi_data *data = dev->data;
+
+	config->irq_configure();
+
+	/*
+	 * Configure the FIFO control register.
+	 * Set the threshold as 1 for getting notification immediately
+	 * on converting completed.
+	 */
+	hw_write_masked(&ADC_RPI_HW->fcs, 1 << ADC_FCS_THRESH_LSB, ADC_FCS_THRESH_BITS);
+	hw_set_bits(&ADC_RPI_HW->fcs, ADC_FCS_ERR_BITS);
+	hw_set_bits(&ADC_RPI_HW->fcs, ADC_FCS_SHIFT_BITS);
+	hw_set_bits(&ADC_RPI_HW->fcs, ADC_FCS_EN_BITS);
+
+	/* Set max speed to conversion */
+	ADC_RPI_HW->div = 0;
+
+	/* Enable ADC and wait becoming READY */
+	ADC_RPI_HW->cs = ADC_CS_EN_BITS;
+	while (!(ADC_RPI_HW->cs & ADC_CS_READY_BITS))
+		;
+
+	/* Enable FIFO interrupt */
+	ADC_RPI_HW->inte = 1U;
+
+	adc_context_unlock_unconditionally(&data->ctx);
+
+	return 0;
+}
+
+#define IRQ_CONFIGURE_FUNC(inst)					     \
+	static void adc_rpi_configure_func_##inst(void)			     \
+	{								     \
+		IRQ_CONNECT(DT_INST_IRQN(inst), DT_INST_IRQ(inst, priority), \
+			    adc_rpi_isr, DEVICE_DT_INST_GET(inst), 0);	     \
+		irq_enable(DT_INST_IRQN(inst));				     \
+	}
+
+#define IRQ_CONFIGURE_DEFINE(inst) .irq_configure = adc_rpi_configure_func_##inst,
+
+#define ADC_RPI_INIT(_num)							   \
```

Comment:
For the other drivers we've named this `idx`

Reply:
I still see `_num`, is it just the diff not working correctly?

Reply:
I forget update it.
It is correct already when now.

Reply:
> I forget update it.
> It is correct already when now.

I am confused by this response - do you mean you will update it to use `idx`?

Reply:
@mbolivar-nordic 

Sorry for the confusion.

I made the fix on June 9.
But the repository was not updated because the push failed.
I uploaded the fix just a few minutes ago.
This comment fix this comment and the following two remarks with this update.

https://github.com/zephyrproject-rtos/zephyr/pull/45363#discussion_r892171981
https://github.com/zephyrproject-rtos/zephyr/pull/45363#discussion_r892177355